src/number_theory/multiplicity.lean

```diff
@@ -0,0 +1,412 @@
+/-
+Copyright (c) 2022 Tian Chen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tian Chen, Mantas Bakšys
+-/
+import algebra.geom_sum
+import data.int.parity
+import number_theory.padics.padic_norm
+import data.zmod.basic
+
+/-!
+# Multiplicity in Number Theory
+
+This file contains results in number theory relating to multiplicity.
+
+## Main statements
+
+* `multiplicity.int.pow_sub_pow` is the lifting the exponent lemma for odd primes.
+  We also prove several variations of the lemma.
+
+## References
+
+* [Wikipedia, *Lifting-the-exponent lemma*]
+  (https://en.wikipedia.org/wiki/Lifting-the-exponent_lemma)
+-/
+
+open ideal ideal.quotient finset
+open_locale big_operators
+
+variables {R : Type*} {n : ℕ}
+
+
+section comm_ring
+variables [comm_ring R] {a b x y : R}
+
+
+lemma dvd_geom_sum₂_iff_of_dvd_sub {x y p : R} (h : p ∣ x - y) :
+  p ∣ ∑ i in range n, x ^ i * y ^ (n - 1 - i) ↔ p ∣ n * y ^ (n - 1) :=
+begin
+  rw [← mem_span_singleton, ← ideal.quotient.eq] at h,
+  simp only [← mem_span_singleton, ← eq_zero_iff_mem, ring_hom.map_geom_sum₂, h, geom_sum₂_self,
+    _root_.map_mul, map_pow, map_nat_cast]
+end
+
+lemma dvd_geom_sum₂_iff_of_dvd_sub' {x y p : R} (h : p ∣ x - y) :
+  p ∣ ∑ i in range n, x ^ i * y ^ (n - 1 - i) ↔ p ∣ n * x ^ (n - 1) :=
+by rw [geom_sum₂_comm, dvd_geom_sum₂_iff_of_dvd_sub]; simpa using (dvd_neg _ _).mpr h
+
+lemma dvd_geom_sum₂_self {x y : R} (h : ↑n ∣ x - y) : ↑n ∣ ∑ i in range n, x ^ i * y ^ (n - 1 - i):=
+(dvd_geom_sum₂_iff_of_dvd_sub h).mpr (dvd_mul_right _ _)
+
+lemma sq_dvd_add_pow_sub_sub (p x : R) (n : ℕ) :
+  p ^ 2 ∣ (x + p) ^ n - x ^ (n - 1) * p * n - x ^ n :=
+begin
+  cases n,
+  { simp only [pow_zero, nat.cast_zero, mul_zero, sub_zero, sub_self, dvd_zero]},
+  { simp only [nat.succ_sub_succ_eq_sub, tsub_zero, nat.cast_succ, add_pow,
+      finset.sum_range_succ, nat.choose_self, nat.succ_sub _, tsub_self, pow_one,
+      nat.choose_succ_self_right, pow_zero, mul_one, nat.cast_zero, zero_add, nat.succ_eq_add_one],
+    abel,
```

Comment:
Does `abel` close a goal here? If so you should use indentation to make clear there several goals.

Reply:
Abel does not close the goal there

Reply:
`abel` is a "fancy" tactic because the resulting goal depends on the order the variables are declared in. So you should not use a rigid tactic like `apply` afterwards.